src/com/tigersale/userInterface/customerUser/ListOfProductView.java

```diff
@@ -0,0 +1,195 @@
+                //Print all the products with associated numbers
+                System.out.println("Please choose the number associated with the product to view detail:");
+                int i = 0;
+                while (i < size) {
+                    System.out.println((i+2) + ": " + products.get(i).name);
+                    i++;
+                }
+
+
+                // Try to get a numeric response from the user
+                try {
+                    choice = scanner.nextInt();
+                } catch (InputMismatchException e) {
+                    System.out.println("Please type in an integer corresponding to your preferred option.");
+                    scanner.next();
+                    continue;
+                }
+
+                //Invalid option
+                if (choice > (size + 1))
+                {
+                    System.out.println("I am sorry, the option you chose does not exist. Please try again.");
+                    continue;
+                }
+                //Valid option, view detail of the chosen product
+                if (choice == 0)
+                {
+                    return;
+                } else if (choice == 1)
+                {
+                    (new ShoppingListView(scanner, user)).runShoppingListView();
+                } else
+                {
+                    //Product's index = choice - 2 because we have used 0 and 1 for other options above
+                    Product chosen = products.get(choice-2);
+                    viewProductDetail(chosen);
+                }
+            }
+        }
+
+    }
+
+    /**
+     * This method acts as a Product Detail view for CustomerUser
+     * @param chosen the chosen Product to view detail
+     */
+    private void viewProductDetail(Product chosen) {
+        while (true)
+        {
+            System.out.println("=========================================================");
+            System.out.println("                     Product Detail                      ");
+            System.out.println("=========================================================");
+            System.out.println(chosen.toString());
+            System.out.println("------------------------------------------------------------");
+            System.out.println("Do you want to add this product to your shopping list?");
+            System.out.println("0: No. Bring me back to Product List");
+            System.out.println("1: Yes");
+
+
+            int choice;
+            // Try to get a numeric response from the user
+
+            try
+            {
+                choice = scanner.nextInt();
+            } catch (InputMismatchException e)
+            {
+                System.out.println("Please type in an integer corresponding to your preferred option.");
+                scanner.next();
+                continue;
+            }
+            //Invalid option:
+            if (choice > 1)
```

Comment:
Use switch statement here.  You can do 
switch(choice)
{
case 0:
...
break;
case 1:
...
break;
default:
// this is your error case or "choice > 1"
break;